### ISSUE TITLE
curl: (6) Could not resolve host: www.duckdns.org

### ISSUE THREAD

**Issue by chgaertner** [linuxserver/docker-duckdns#34]
[linuxserverurl]: https://linuxserver.io
[![linuxserver.io](https://raw.githubusercontent.com/linuxserver/docker-templates/master/linuxserver.io/img/linuxserver_medium.png)][linuxserverurl]

The update of the DuckDNS IP address fails because the host "www.duckdns.org" could not be resolved (needed in the script "/app/duck.sh"). I tried to find a solution online but did not succeed.

------------------------------

## Expected Behavior
The script /app/duck.sh should update the DuckDNS IP address for my subdomain.

## Current Behavior
The update fails because the host "www.duckdns.org" could not be resolved.

## Steps to Reproduce
1. Install the actual Docker software on a Raspberry Pi 4 with the actual RaspiOS.
2. Use the "docker-compose.yml" from this site.
3. Start the container with "docker-compose up -d"
4. Wait for the DuckDNS IP address to be updated.
5. Exec into the container, start the command from /app/duck.sh without the silent parameter (-s).
6. Error message "curl: (6) Could not resolve host: www.duckdns.org"

## Environment
**OS:** RaspiOS (2021-01-11-raspios-buster-armhf-lite)
**CPU architecture:** arm32
**How docker service was installed:** Using the official "get-docker.sh" script from https://get.docker.com

## Command used to create docker container (run/create/compose/screenshot)
docker-compose.yml
version: '2.1'
services:
  duckdns:
    image: ghcr.io/linuxserver/duckdns
    container_name: duckdns-cg
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=Europe/Vienna
      - SUBDOMAINS=****
      - TOKEN=****
      - LOG_FILE=true
    volumes:
      - ./config:/config
    restart: unless-stopped

## Docker logs
User uid:    1000
User gid:    1000
[cont-init.d] 10-adduser: exited 0.
[cont-init.d] 40-config: executing...
Retrieving subdomain and token from the environment variables
log will be output to file
Something went wrong, please check your settings  Thu Feb 4 16:06:40 CET 138072
[cont-init.d] 40-config: exited 0.
[cont-init.d] 99-custom-files: executing...
[custom-init] no custom files found exiting...
[cont-init.d] 99-custom-files: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.

Comment: Today is 2021-03-12, the message "Something went wrong, please check your settings  Thu Feb 4 16:06:40 CET 138072" above seems strange. The date/time of my docker host is set correctly.

**Comment by Roxedus** [comment]
Can you post full logs please

**Comment by chgaertner** [comment]
I am sorry, I missed the beginning of the logs due to the lsio-logo which I erroneously considered to be the start of the log.

Greetings, Christian


[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 01-envfile: executing...
[cont-init.d] 01-envfile: exited 0.
[cont-init.d] 10-adduser: executing...

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

Your DockerHost is most likely running an outdated version of libseccomp

To fix this, please visit https://docs.linuxserver.io/faq#libseccomp

Some apps might not behave correctly without this

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@


-------------------------------------
          _         ()
         | |  ___   _    __
         | | / __| | |  /  \
         | | \__ \ | | | () |
         |_| |___/ |_|  \__/


Brought to you by linuxserver.io
-------------------------------------

To support the app dev(s) visit:
DuckDNS: https://www.patreon.com/user?u=3209735

To support LSIO projects visit:
https://www.linuxserver.io/donate/
-------------------------------------
GID/UID
-------------------------------------

User uid:    1000
User gid:    1000
-------------------------------------

[cont-init.d] 10-adduser: exited 0.
[cont-init.d] 40-config: executing...
Retrieving subdomain and token from the environment variables
log will be output to file
Something went wrong, please check your settings  Thu Feb 4 16:06:40 CET 138072
[cont-init.d] 40-config: exited 0.
[cont-init.d] 99-custom-files: executing...
[custom-init] no custom files found exiting...
[cont-init.d] 99-custom-files: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.

**Comment by Roxedus** [comment]
Look above the logo :)

**Comment by chgaertner** [comment]
Do you mean the statement about the outdated version of libseccomp? I did come across the page "https://docs.linuxserver.io/faq#libseccomp" when searching for a solution on the web. But I did not consider this to be the reason for my problem because none of the items listed in the Symptoms section seemed to be applicable.

I re-read the article and try to get the point.
- Option 1 does not apply because I do not use Plex or Jellyfin.
- Option 2 and Option 3 might be helpful.
- Option 4 was not taken into account because I use the newest OS version (dated 2021-01-11) and does still get updates.

Do I get you correct that you recommend to update libseccomp?

Greetings, Christian

**Comment by Roxedus** [comment]
You are hitting the last symptom.

**Comment by chgaertner** [comment]
Okay, I would have never realised this without your help. Thank you very much!

I will try Option 2 and update the library. Which of these library versions do you recommend? 2.5.1-1 is the newest but the other one sounds like "backport" and therefore might be better?

libseccomp-dev_2.4.4-1~bpo10+1_armhf.deb	2020-09-26 15:13	70K
libseccomp-dev_2.5.1-1_armhf.deb		2020-12-21 10:28	84K

**Comment by Roxedus** [comment]
2.5.1 might work.

**Comment by chgaertner** [comment]
I just tried to install 2.5.1 but I got some errors; however, version 2.4.4-1~bpo10+1 was successfully installed. I restarted the container and - drum roll - the DuckDNS IP has been updated. Everything works like expected now.

As a reference, I used these commands:
wget http://ftp.us.debian.org/debian/pool/main/libs/libseccomp/libseccomp2_2.4.4-1~bpo10+1_armhf.deb
sudo dpkg -i libseccomp2_2.4.4-1~bpo10+1_armhf.deb

Again, thank you very much for your quick and qualified support!

Greetings from Vienna/Austria to Norway! :-)
Christian
